config/config.go

```diff
@@ -234,27 +235,27 @@ func loadExtensions(v *viper.Viper, factories map[string]extension.Factory) (con
 	// Iterate over extensions and create a config for each.
 	for key := range keyMap {
 		// Decode the key into type and fullName components.
-		typeStr, fullName, err := decodeTypeAndName(key)
-		if err != nil || typeStr == "" {
```

Comment:
My understanding is that to fix the problem that you found we just needed to delete the `|| typeStr == ""` and ensure `DecodeTypeAndName` doesn't return empty typeStr together with nil err (which it doesn't) and perhaps add a comment to `DecodeTypeAndName` that explains it is part of the contract.

The rest of the changes are not required to fix this potential bug. Optionally we can refactor DecodeTypeAndName to return a struct instead of pairs of strings. That would be a slight improvement which ensures one cannot confuse the order of return values. The rest of the changes looks unrelated to me and is best to be left out from this PR.

Reply:
Ok will leave all that out for now, agree with your other comments above.